### PR TITLE
feat(content-sidebar): Add placehoder metadatasidebar redesigned

### PR DESCRIPTION
**Pull request by jankowiakdawid**
This PR introduces new placeholder component for work on redesigning Metadata Sidebar.
At current state this placeholder component doesn't offer much functionality nor look.
Screenshot 2024-07-03 at 23 51 03

The idea is to have this component to align and parallel work on upgrading MetadataSidebar to use Blueprint-web components and extends it's functionality with AI.

The component is being lazy loaded to not add to main bundle, and is hidden behind additional feature flag.

### CODE REVIEW

File: src/elements/content-sidebar/SidebarPanels.js
Comment (jstoffan) [inline src/elements/content-sidebar/SidebarPanels.js:203]: Let's use the `isFeatureEnabled` utility here.

**Review by jstoffan — COMMENTED** [review]
Do we have a feature branch we want to use? Or do we intend to merge to master until we're blocked by internal dependencies?

File: src/elements/content-sidebar/MetadataSidebarRedesigned.tsx
Comment (jankowiakdawid) [reply src/elements/content-sidebar/MetadataSidebarRedesigned.tsx:9]: With recent update in master they no longer necessary. I've removed them after the update.

**Comment by jankowiakdawid** [comment]
> Do we have a feature branch we want to use? Or do we intend to merge to master until we're blocked by internal dependencies?

We don't have yet. I was hoping we can get this one to `master` and then spawn a feature branch for us to merge to. If that is not advised I'll create a new branch right away.

File: src/constants.js
Comment (JChan106) [inline src/constants.js:399]: can we have this named `metadata_redesign` to match the variable name and a consistent naming convention

**Review by JChan106 — DISMISSED** [review]
lgtm, lets spawn a feature branch after merging as mentioned

File: src/constants.js
Comment (tjuanitas) [inline src/constants.js:318]: this constant is the same value as the one above, is that okay? will there be conflicts anyway such as if these are used in a switch statement?

if this is intended, should we reuse the existing constant so it won't need to be changed back later?

File: src/elements/content-sidebar/MetadataSidebarRedesign.tsx
Comment (tjuanitas) [inline src/elements/content-sidebar/MetadataSidebarRedesign.tsx:18]: new classnames should follow SUIT naming:
`bcs-MetadataSidebarRedesign`

File: src/elements/content-sidebar/SidebarPanels.js
Comment (tjuanitas) [inline src/elements/content-sidebar/SidebarPanels.js:117]: since only one version of metadatasidebar can render at a time, you could reuse the ref above for the new redesign so there's less code to clean up

**Review by tjuanitas — DISMISSED** [review]
lgtm

File: src/elements/content-sidebar/MetadataSidebarRedesign.tsx
Comment (jankowiakdawid) [reply src/elements/content-sidebar/MetadataSidebarRedesign.tsx]: Got it, renamed

File: src/elements/content-sidebar/SidebarPanels.js
Comment (jankowiakdawid) [reply src/elements/content-sidebar/SidebarPanels.js]: Good idea, I delete it

File: src/constants.js
Comment (jankowiakdawid) [reply src/constants.js]: I've looked and they are used only for performance marks, I couldn't find where we use those though.
I decided to do the same as with `ref`s and remove this one, and reuse the `ORIGIN_METADATA_SIDEBAR`

File: src/elements/content-sidebar/SidebarPanels.js
Comment (tjuanitas) [inline src/elements/content-sidebar/SidebarPanels.js:82]: `ORIGIN_METADATA_SIDEBAR_REDESIGN` no longer exists - was previously the same as `ORIGIN_METADATA_SIDEBAR`
```suggestion
const MARK_NAME_JS_LOADING_METADATA_REDESIGNED = `${ORIGIN_METADATA_SIDEBAR}${BASE_EVENT_NAME}`;
```

Comment (jankowiakdawid) [reply src/elements/content-sidebar/SidebarPanels.js:82]: After further consideration I decided to leave `ORIGIN_METADATA_SIDEBAR_REDESIGN` and apply it properly, so we can track the loading performance. I'm sure @jstoffan knows how.

**Review by JChan106 — APPROVED** [review]
lgtm

**Review by tjuanitas — APPROVED** [review]

**Pull request merged**
